drafts/draft-0004.md

```diff
@@ -0,0 +1,160 @@
+
+## Regulated Assets Flow
+The following illustrates a transaction submission flow for regulated assets.
+
+## Stellar.toml
+Issuers will advertise the existence of an Approval Service through their stellar.toml file. This is done in the [[CURRENCIES]] section as different assets can have different requirements.
+
+## Fields:
+- `regulated` is a boolean indicating whether or not this is a regulated asset.
+- `validaton_server` is the url of a validations service that performs compliance checks on transactions and if necessary, modifies them to enforce compliance.
+- `approval_server` is the url of an approval service and identifies this is a Regulated Asset. 
+- `approval_criteria` is a human readable string that explains the issuer's requirements for validating and signing transactions.
+
+### Example
+```toml
+[[CURRENCIES]]
+code="GOAT"
+issuer="GD5T6IPRNCKFOHQWT264YPKOZAWUMMZOLZBJ6BNQMUGPWGRLBK3U7ZNP"
+regulated=true
+validation_server="https://goat.io/tx_validate"
+approval_server="https://goat.io/tx_auth"
+approval_criteria= “The goat approval server will ensure that transactions are compliant with NFO regulation”
+```
+
+## Validation Server
+The transaction validation server receives unsigned transactions and checks for compliance. It can choose to accept the transaction, modify it to fit compliance, or reject it. 
+Request 
+HTTP POST request is encoded as application/json.
+
+Parameters :
+Name
+Data Type
+Description
+tx
+string
+Unsigned Transaction XDR, Base64 encoded.
+This transaction will be tested for compliance.
+Responses
+HTTP responses are encoded as application/json
+Success (status code 200)
+A successful response means that the transaction is compliant or the validation service was able to modify it to meet compliance.
+
+Parameters :
+Name
+Data Type
+Description
+tx
+string
+Unsigned Transaction XDR, Base64 encoded.
+This transaction is compliant.
+message
+string
+A human readable string containing information about the changes. made to the transaction in order to make it compliant (optional).
+Pending (status code 202)
+A pending response means that the issuer needs to asynchronously validate the transaction. The user should resubmit the transaction after a given timeout. 
+
+Parameters:
+Name
+Data Type
+Description
+timeout
+integer
+Number of milliseconds to wait before submitting the same transaction
+message
+string
+A human readable string containing information about the reason 
+Rejected (status code 400)
+A rejection response means that the transaction is not compliant and could not be made compliant.
+
+Parameters:
+Name
+Data Type
+Description
+error
+string
+A human readable string explaining why the transaction is not compliant and could not be made compliant.
```

Comment:
Keep in mind that most of the time, regulated agencies require that the reasons for rejections of funds transfers are for the financial institution itself, but not for consumption by the account holder.  This is for security and other reasons.  So, if you fail a compliance check, the "reason" returned from here would probably be "You should contact your issuer."